scripts/stats.py

```diff
@@ -3,83 +3,269 @@
 import collections
 import itertools
 import pathlib
+import argparse
+from datetime import datetime
 
+Path = collections.namedtuple("Entry", ("suffix", "group", "name"))
+avoid_extensions = [
+    "",
+    ".md",
+    ".png",
+    ".csv",
+    ".class",
+    ".data",
+    ".in",
+    ".jpeg",
+    ".jpg",
+    ".out",
+    ".textclipping",
+    ".properties",
+    ".txt",
+    ".sbt",
+]
+avoid_dirs = ["project", "test", "img", "image", "images"]
+paths = []
+original_paths = []
+for path in pathlib.Path(__file__).parents[1].glob("code/**/**/*"):
+    if (
+        path.suffix
+        and not any(elem in list(path.parts) for elem in avoid_dirs)
+        and path.suffix.lower() not in avoid_extensions
+    ):
+        original_paths.append(path.parts)
+        paths.append(
+            Path(
+                suffix=path.suffix.lstrip(".").lower(),
+                group=path.parts[1].replace("-", " ").replace("_", " "),
+                name=path.parts[-2].replace("-", " ").replace("_", " "),
+            )
+        )
 
-Path = collections.namedtuple('Entry', ('suffix', 'group', 'name'))
+suffixes = {path.suffix for path in paths}
+suffixes = sorted(suffixes - set(avoid_extensions))
+extension_map = {
+    "adb": "Ada",
+    "bf": "BrainFuck",
+    "c": "C",
+    "clj": "Clojure",
+    "cpp": "C++",
+    "cr": "CRiSP",
+    "cs": "C#",
+    "elm": "ELM",
+    "erl": "Erlang",
+    "ex": "Euphoria",
+    "exs": "Elixir",
+    "f": "Fortran",
+    "fs": "Visual F#",
+    "go": "Golang",
+    "h": "C",
+    "hpp": "C++",
+    "hs": "Haskell",
+    "htm": "HTML",
+    "html": "HTML",
+    "ipynb": "Jupyter Notebook",
+    "java": "Java",
+    "jl": "Julia",
+    "js": "JavaScript",
+    "kt": "Kotlin",
+    "lua": "Lua",
+    "m": "Objective-C",
+    "ml": "ML",
+    "nim": "Nim",
+    "pde": "Processing Development Environment",
+    "php": "PHP",
+    "pl": "Perl",
+    "purs": "PureScript",
+    "py": "Python",
+    "rb": "Ruby",
+    "re": "Reason",
+    "rkt": "Racket",
+    "rs": "Rust",
+    "ruby": "Ruby",
+    "sc": "SuperCollider",
```

Comment:
The one file with this extension should be a `.scala` file. Please fix that in this PR. SuperCollider is an audio format if I researched correctly.